### PR TITLE
Upgrade nette

### PR DESCRIPTION
**Pull request by VBoss**
Composer require PHP >= 7.1

Travis and Composer cleanup
- Unified travis.yml
- Added PHPStan to travis
- Added CodingStandard to travis
- removed unused nette libraries
- Added PHPStan
- Added Kdyby/CodingStandard
- Moved PHPCoveralls to composer
- Added Typo3/ClassAliasLoader

Exception split to own files
- Old namespaces aliased

Other
- Local php ini for tests
- Removed unused register method
- Nette framework version resolving not to be dependent on nette/application
- fixed onShutdown event for PHPStan `onShutdown` only magically calls foreach on property, so i did it here too.
- \Kdyby\StrictObjects\Scream is not compatible with Nette/SmartObject

unused arguments in tests reported by PHPStan 
- so use them as if it was application
- added to properties and made getters

TODO 
- needs kdyby/events update to nette 3.0

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -17,22 +17,28 @@
...
+		"nette/http": "^3.0",
+		"kdyby/events": "^3.2@dev",
+		"symfony/event-dispatcher": "~2.3 || ^3.0 || ^4.0",
+
+		"nette/tester": "^2.2",
+        "phpstan/phpstan-shim": "^0.11.5",
+        "kdyby/coding-standard": "^1.0",
+        "php-coveralls/php-coveralls": "^2.1",
+        "jakub-onderka/php-parallel-lint": "^1.0",
+        "typo3/class-alias-loader": "^1.0"
```
Comment (fprochazka) [inline composer.json]: maličko ti to ujely whitespaces

File: src/Exception/InvalidArgumentException.php [hunk 2]
```diff
@@ -0,0 +1,9 @@
+<?php declare(strict_types = 1);
+
+namespace Kdyby\Console\Exception;
+
+
+class InvalidArgumentException extends \InvalidArgumentException implements \Kdyby\Console\Exception\Exception
+{
+
+}
```
Comment (fprochazka) [inline src/Exception/InvalidArgumentException.php]: ty prazdny řadky na konci by byly taky fajn

File: composer.json [hunk 3]
```diff
@@ -17,22 +17,28 @@
...
+		"symfony/event-dispatcher": "~2.3 || ^3.0 || ^4.0",
+
+		"nette/tester": "^2.2",
+		"phpstan/phpstan-shim": "^0.11.5",
+		"kdyby/coding-standard": "^1.0",
+		"php-coveralls/php-coveralls": "^2.1",
+		"jakub-onderka/php-parallel-lint": "^1.0",
+		"typo3/class-alias-loader": "^1.0"
 	},
-	"minimum-stability": "dev",
```
Comment (fprochazka) [inline composer.json]: tohle by tady imho mělo zustat, ne?

Comment (VBoss) [reply composer.json]: ok vrátím

**Pull request merged**